### PR TITLE
refactor!: Split AmbientTransactionOptions in SpannerTransactionCreationOptions and SpannerTransactionOptions.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
In preparation for using SpannerTransactionCreationOptions and SpannerTransactionOptions for all transaction types.

BREAKING CHANGE: The AmbientTransactionOptions class is split in SpannerTransactionCreationOptions and SpannerTransactionOptions, with all original options moving to one or the other. The SpannerConnection.Open set of methods that received an AmbientTransactionOptions before now receive one SpannerTransactionCreationOptions and one SpannerTransactionOptions.

Note that AmbientTransactionOptions has only ever been released on beta versions of the library, so we expect the impact of this breaking change to be low.

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
Closing and reopening to allow breaking changes.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// Options that may be applied to a transaction after creation, usually before commiting the transaction
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs:21]: Nit: commiting => committing

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+/// or before executing the first transactional statement.
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs:22]: Possibly explain that this is mutable (compared with SpannerTransactionCreationOptions) and under what circumstances mutations are observed? Or is the plan to put this in the conceptual docs?

**Review by jskeet — APPROVED** [review]
Just a few nits and suggestions, but I'm happy with the overall idea.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionCreationOptions.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// Options that may be provided when creating a <see cref="SpannerTransaction"/>.
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionCreationOptions.cs:20]: Yes, done

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionCreationOptions.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+/// </summary>
+public sealed class SpannerTransactionCreationOptions
+{
+    /// <summary>
+    /// Default options for creating a transaction. Using these options will result in a read-write transaction.
+    /// </summary>
+    public static SpannerTransactionCreationOptions Default { get; } = new SpannerTransactionCreationOptions(timestampBound: null, transactionId: null);
+
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionCreationOptions.cs:28]: Done

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs (hunk 2 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs:22]: I'll add here, even if we add some of this elsewhere, it doesn't harm to have it in here.

**Review by amanda-tarafa — COMMENTED** [review]
Thanks, I'll push changes shortly and merge on green.

**Pull request merged**